### ISSUE TITLE
Linux command-line installer offers invalid option

### ISSUE THREAD

**Issue by BarryTice** [zen-browser/desktop#1651]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

I am installing the "generic" version in Ubuntu (20.04), though I didn't quite follow directions. Rather than curling the bash script and directing it to bash, I downloaded the file (in order to review it), set the execute bit, and ran it directly. (This does not affect the nature of this bug.)

When you launch the `appimage.sh` script to install the appimage that you've already downloaded, you get this list of options, which says that your choices are 0 through 3. However `3` is clearly not a valid option.

```
[i] (★^O^★) What would you like to do?
[i]   1) Install
[i]   2) Uninstall
[i]   0) Exit
Enter your choice (0-3): 3
[e] (•ˋ _ ˊ•) Invalid choice. Exiting...
```

If `3` is not a valid choice, then the `Enter your choice (0-3):` message should only say `Enter your choice (0-2):`

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.3

### What platform are you seeing the problem on?

Linux

### Relevant log output

```shell
barry@barry-bob:~/Downloads$ ./appimage.sh 
[i] ╔════════════════════════════════════════════════════╗
[i] ║                                                    ║
[i] ║    (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧  Zen Browser Installer            ║
[i] ║                                                    ║
[i] ║    CPU: AVX2 Supported (Optimized Version)         ║
[i] ║    Status: Zen Browser Installed                   ║
[i] ║    zsync: Not Installed (Needed for Updates)       ║
[i] ║                                                    ║
[i] ╚════════════════════════════════════════════════════╝
[i] 
[i] (★^O^★) What would you like to do?
[i]   1) Install
[i]   2) Uninstall
[i]   0) Exit
Enter your choice (0-3): 3
[e] (•ˋ _ ˊ•) Invalid choice. Exiting...
```

**Comment by BarryTice** [comment]
Apparently `3` is valid if I have `zsync` installed, which, apparently, I don't.
```
main_menu() {
    log_info "(★^O^★) What would you like to do?"
    log_info "  1) Install"
    log_info "  2) Uninstall"
    if check_zsync_installed; then
        log_info "  3) Check for Updates"
    fi
    log_info "  0) Exit"
    read -p "Enter your choice (0-3): " main_choice
```
In any case, the `if` block at line 276 should conditionally set the "last number in list" to 3 if there is a 3 option, and then line 280 should use that variable rather than the hard-coded `3` value.

**Comment by OffShur3** [comment]
Yes, you should install zsync for this option.

**Comment by BarryTice** [comment]
Certainly, with zsync installed this presents no issue. However without zsync installed the installer gives incorrect information.

**Comment by Samueru-sama** [comment]
Note that the zsync file is not being used for delta updates and only to [check if an update is available](https://github.com/zen-browser/desktop/pull/1435#issuecomment-2365432753). 

I personally don't use it since I use [AppMan](https://github.com/ivan-hc/AM) to install AppImages, and in this case you also get delta updates when you install appimageupdate with AppMan (which is what the zsync file is for), that is it doesn't download the entire appimage when updating.

**Comment by BarryTice** [comment]
Clearly zsync isn't a **required** dependency. I fully accept that _if_ zsync is present, the installer bash script works as it should, presenting options 0 through 3, with all being valid.

My issue is that when zsync is not present, the installer still says that the range of valid inputs is 0 to 3, but then tells you that 3 isn't a valid option if selected.

Is this pretty much the working definition of "corner case"? Yes. Is this also some pretty reasonable bounds checking on inputs? Yes, it is. but "no zsync" is a reasonable expectation, and then the installer script presents incorrect information about what the range of valid inputs (for the context) is.

**Comment by LeMoonStar** [comment]
The AppImage install script has multiple issues. The update functionality (Option 3, falsely offered in this case) itself is also broken, even if ZSync is installed (#2169).
The script currently doesn't seem to be a priority and is unlikely to be fixed any time soon unless somebody is willing to do it themselves. Especially it's improper usage of ZSync makes it less than optimal.

**Comment by BarryTice** [comment]
I understand that script fixes are not a priority, but as a software QA professional I have _all kinds of issues_ with issue tickets being closed simply because they've been open for less than three months.

The fix for this would be trivial to implement, in theory: check if `zsync` is present (since it is explicitly NOT a requirement and even Samueru-sama says that they don't use it); then adjust the input prompt to read "0-2" instead of "0-3" if it's not there.
